lib/rails_values/blank_country.rb

```diff
@@ -9,7 +9,8 @@ def <=>(other)
     end
 
     def eql?(other)
-      alpha2 == other.alpha2
+      other.is_a?(BlankCountry) ||
+        (other.is_a?(Country) && alpha2 == other.alpha2)
```

Comment:
it's only equal if the `other` would also be a BlankCountry

```suggestion
      Country.cast(other).is_a?(BlankCountry)
```